src/components/SchedulePanel/index.js

```diff
@@ -0,0 +1,361 @@
+      this.setState({ 
+        description: this.scheduleControlInfo.Description ? 
+        this.scheduleControlInfo.Description.replace(new RegExp("SlSh", 'g'), '/').replace(new RegExp("CoLn", 'g'), ':')
+      .replace(new RegExp("dOT", 'g'), '.') : ''});
+      this.setState({ modifiedTimeStamp: this.scheduleControlInfo.ModifiedTimeStamp });
+    }
+    console.log('Schedule Control : ', this.scheduleControlInfo);
+    this.setState({ loading: false });
+  };
+
+  handleOfficeHoursOpen = slotInfo => {
+    this.setState({ officeHoursOpen: true });
+    if (slotInfo){
+      let startTime = slotInfo.start.toTimeString().split(":");
+      let endTime = slotInfo.end.toTimeString().split(":");
+      this.setState({start:startTime[0] + ":" + startTime[1],
+        end:endTime[0] + ":" + endTime[1], resourceId: slotInfo.resourceId,
+      });
+    }
+    this.setState({ isDoubleClick: false });
+  };
+
+  handleOfficeHoursClose = () => {
+    this.setState({
+      officeHoursOpen: false,
+      isDoubleClick: false,
+      selectedEvent: null,
+    });
+  };
+
+  handleRemoveOfficeHoursOpen = () => {
+    this.setState({ removeOfficeHoursOpen: true });
+  };
+
+  handleRemoveOfficeHoursClose = () => {
+    this.setState({ removeOfficeHoursOpen: false });
+  };
+
+  handleRemoveButton = event => {
+    if (event.id > 1000) {
+      this.setState({ disabled: false, selectedEvent: event }, () => {
+        console.log(this.state.selectedEvent)});
+    } else {
+      this.setState({ disabled: true});
+    }
+  };
+
+  handleEditDescriptionOpen = () => {
+    this.setState({ editDescriptionOpen: true });
+  };
+
+  handleEditDescriptionClose = () => {
+    this.setState({ editDescriptionOpen: false });
+  };
+
+  handleEditDescriptionButton = () => {
+    this.setState({ disabled: false });
+  };
+
+  handleDescriptionSubmit = async descValue => {
+    await schedulecontrol.setScheduleDescription(descValue);
+    window.location.reload(); // refresh to show the change
+  };
+
+  handleHoursSubmit = mySchedule => {
+    var data = {
+      Event_ID: null,
+      Gordon_ID: this.props.profile.ID,
+      DESCRIPTION: mySchedule.description,
+      LOCATION: mySchedule.location,
+      MON_CDE: mySchedule.monday ? 'M' : null,
+      TUE_CDE: mySchedule.tuesday ? 'T' : null,
+      WED_CDE: mySchedule.wednesday ? 'W' : null,
+      THU_CDE: mySchedule.thursday ? 'R' : null,
+      FRI_CDE: mySchedule.friday ? 'F' : null,
+      SAT_CDE: mySchedule.saturday ? 'S' : null,
+      SUN_CDE: mySchedule.sunday ? 'N' : null,
+      IS_ALLDAY: mySchedule.allDay ? 1 : 0,
+      BEGIN_TIME: mySchedule.startHour,
+      END_TIME: mySchedule.endHour,
+    };
+
+    if (this.state.isDoubleClick) {
+      this.setState({ isDoubleClick: false });
+      data.Event_ID = this.state.selectedEvent.id;
+      myschedule
+        .updateMySchedule(data)
+        .then(value => {
+          window.location.reload();
```

Comment:
Surely there's a better way to re-render the component than reloading the whole page?

Reply:
Change made